[TDX-1630] OIDC: share email as Admin and Dev (#3388)
Kong/docs.konghq.com: app/_hub/kong-inc/openid-connect/index.md

  4. `config.audience_required` (if using a public identity provider)
  5. `config.session_secret` (if using the Kong in DB-less mode)

  ## Authenticate Dev Portal with OIDC

>   You can authenticate a self-managed Dev Portal with this plugin.

Reviewer: ```suggestion
  You can [authenticate a self-managed Dev Portal](/gateway/latest/developer-portal/configuration/authentication/oidc/) with this plugin.
```

>   The OIDC plugin enables you to use the same email address for

Reviewer: We have a bunch of documents for Dev Portal that talk about setting up OIDC and email verification. You should add this info to at least this topic: 
https://docs.konghq.com/gateway/2.6.x/developer-portal/configuration/authentication/oidc/

I'd also recommend looking through the on-prem dev portal docs to see if info about OIDC and Dev Portal appears anywhere else. I didn't see anything at first glance, but it's worth checking.

Author: Aha, great. I will add a note here too.

Author: Checked the on-prem dev portal docs and I also didn't find that information. I do think it makes more sense to keep this content in both Enable OpenID Connect in the Dev Portal and the OIDC plugin docs.

>   both your Admin and your Developer accounts, in case you want to have the same
>   credentials between the Admin GUI and your self-managed Dev Portal. To use the

Reviewer: ```suggestion
  both your Admin and your Developer accounts, in case you want to have the same
  credentials between the Admin GUI (Kong Manager) and your self-managed Dev Portal. To use the
```

  same email for both accounts, ensure the Admin `username` is not set to the
  account email and that instead, the `custom_id` is set to the account email.

>   By default, the Admin GUI and the self-managed Dev Portal require distinct email

Reviewer: ```suggestion
  By default, Kong Manager and the self-managed Dev Portal require distinct email
```
